lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java

```diff
@@ -5,11 +5,22 @@
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.nio.charset.StandardCharsets;
-
-public final class Parser implements AutoCloseable {
-  private static void initializeLibraries() {
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public final class Parser {
+  static AtomicBoolean initialized = new AtomicBoolean(false);
+
+  private static void ensureInitialized() {
+    // Optimization: Skip finding and initializing the library when it's definitely already been
+    // done.
+    // Note that this inexpensive check may produce false negatives: Nothing prevents multiple
+    // threads from proceeding to library initialization concurrently (before any thread has
+    // completed initialization), but that is harmless; library initialization is thread-safe and
+    // idempotent.
```

Comment:
I am afraid we will have problems in _native image_ - e.g. `engine-runner/buildNativeImage`. Native image may run initialization during _build time_ - before taking snapshot of the heap. Then (during runtime of the EXE file) the field could already be `true` and no library loaded yet.

Build of `engine-runner/buildNativeImage` is included in the CI - e.g. once we get green, things are ok...